torch/nn/modules/module.py

```diff
@@ -1164,4 +1165,17 @@ def _replicate_for_data_parallel(self):
         replica._parameters = replica._parameters.copy()
         replica._buffers = replica._buffers.copy()
         replica._modules = replica._modules.copy()
+
+        # zero_grad on the replica should zero the parent as well
+        old_zero_grad = replica.__class__.zero_grad
+        weak_self = weakref.ref(replica)
+
+        def zero_grad():
+            replica = weak_self()
+            if replica:
+                old_zero_grad(replica)
+            self.zero_grad()
+
+        replica.zero_grad = zero_grad
```

Comment:
I feel like a more normal way to do this is to just bind the function the normal way: `zero_grad.__get__(self, Module)`

Reply:
I've tried your suggestion but get test failures in `test_data_parallel_model_no_refcycles`. Are you sure that bound methods are weak? I can't find any reference to that in the documentation.

Reply:
Well apparently I'm wrong! (I know there must be something weak involved with bound methods, but I guess maybe not with `__get__`